Compiler bug when declare array with method without returned type

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Ubuntu 21.04
Processor: 12 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
CC version: cc (Ubuntu 10.3.0-1ubuntu1) 10.3.0

getwd: /var/www/vproject/test/testblog
vmodules: /home/tanguy/.vmodules
vroot: /var/www/vproject/v
vexe: /var/www/vproject/v/v
vexe mtime: 2022-01-08 08:05:17
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 c1711b8.ec55c0f

Git version: git version 2.30.2
Git vroot status: weekly.2022.01-3-gec55c0fc
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 3654d6a8

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg /var/www/vproject/test/pattern_framwork/array_function.v`
```v
module main

fn  main() {
	a := [yo()]
}

fn yo() {}
```

**What did you expect to see?**

An error explaining the type of the method is missing.


**What did you see instead?**
```
/var/www/vproject/test/pattern_framwork/array_function.v:4:2: warning: unused variable: `a`
    2 | 
    3 | fn  main() {
    4 |     a := [yo()]
      |     ^
    5 | }
    6 |
==================
/tmp/v_1000/array_function.11669755673098350894.tmp.c:10934: error: declaration of an array of incomplete type elements
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/containers.v
+++ b/vlib/v/checker/containers.v
@@ -85,6 +85,9 @@ pub fn (mut c Checker) array_init(mut node ast.ArrayInit) ast.Type {
 		// }
 		for i, mut expr in node.exprs {
 			typ := c.check_expr_opt_call(expr, c.expr(expr))
+			if typ == ast.void_type {
+				c.error('invalid void array element type', expr.position())
+			}
 			node.expr_types << typ
 			// The first element's type
 			if expecting_interface_array {

--- a/vlib/v/checker/tests/array_init_with_void_value_err.out
+++ b/vlib/v/checker/tests/array_init_with_void_value_err.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/array_init_with_void_value_err.vv:4:8: error: invalid void array element type
+    2 |
+    3 | fn main() {
+    4 |     a := [yo()]
+      |           ~~~~
+    5 |     println(a)
+    6 | }

--- a/vlib/v/checker/tests/array_init_with_void_value_err.vv
+++ b/vlib/v/checker/tests/array_init_with_void_value_err.vv
@@ -1,0 +1,8 @@
+module main
+
+fn main() {
+	a := [yo()]
+	println(a)
+}
+
+fn yo() {}

